src/test/java/io/debezium/connector/vitess/VitessReplicationConnectionIT.java

```diff
@@ -48,6 +48,172 @@ public void beforeEach() {
         TestHelper.execute(TestHelper.INSERT_STMT);
     }
 
+    @Test
+    public void shouldNotErrorOutWhenSkipEnabled() throws Exception {
+        // setup fixture
+        final VitessConnectorConfig conf = new VitessConnectorConfig(
+                TestHelper.defaultConfig(false, false, 1, -1, -1,
+                        null, VitessConnectorConfig.SnapshotMode.NEVER, TestHelper.TEST_SHARD,
+                        "1", "skip").build());
+        final VitessDatabaseSchema vitessDatabaseSchema = new VitessDatabaseSchema(
+                conf, SchemaNameAdjuster.create(), (TopicNamingStrategy) DefaultTopicNamingStrategy.create(conf));
+
+        AtomicReference<Throwable> error = new AtomicReference<>();
+        VitessReplicationConnection connection = new VitessReplicationConnection(conf, vitessDatabaseSchema);
+        Vgtid startingVgtid = Vgtid.of(
+                Binlogdata.VGtid.newBuilder()
+                        .addShardGtids(
+                                Binlogdata.ShardGtid.newBuilder()
+                                        .setKeyspace(conf.getKeyspace())
+                                        .setShard(conf.getShard().get(0))
+                                        .setGtid(Vgtid.CURRENT_GTID)
+                                        .build())
+                        .build());
+
+        BlockingQueue<MessageAndVgtid> consumedMessages = new ArrayBlockingQueue<>(100);
+        AtomicBoolean started = new AtomicBoolean(false);
+        connection.startStreaming(
+                startingVgtid,
+                (message, vgtid, isLastRowEventOfTransaction) -> {
+                    if (!started.get()) {
+                        started.set(true);
+                    }
+                    consumedMessages.add(new MessageAndVgtid(message, vgtid));
+                },
+                error);
+        // Since we are using the "current" as the starting position, there is a race here
```

Comment:
This is a bit unsafe. Could you please take a look at `io.debezium.connector.mysql.StreamingSourceIT.inconsistentSchema(EventProcessingFailureHandlingMode)` to see how to intercept log messages and wait for them?